docs/examples/idealised_flow.py

```diff
@@ -0,0 +1,261 @@
+                U[t, 0, j, i] = (
+                    U0 / (np.cosh(x2 / L) ** 2)
+                    + 2 * U0 * np.sinh(x2 / L) / (np.cosh(x2 / L) ** 3) * G
+                )
+                V[t, 0, j, i] = U0 * L * (1.0 / np.cosh(x2 / L)) ** 2 * G_x
+
+    # Add a linear decay in depth
+    for k in range(1, depth.size):
+        U[:, k, :, :] = U[:, 0, :, :] * ( (depth.size - k) / depth.size)
+        V[:, k, :, :] = V[:, 0, :, :] * ( (depth.size - k) / depth.size)
+
+    # Construct the fieldset
+    data = {"U": U, "V": V, "W": W}
+    dimensions = {"lon": lon, "lat": lat, "depth": depth, "time": times}
+    allow_time_extrapolation = True if len(times) == 1 else False
+    fieldset = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    fieldset.U.interp_method = "cgrid_velocity"
+    fieldset.V.interp_method = "cgrid_velocity"
+    fieldset.W.interp_method = "cgrid_velocity"
+
+    # Add a flat bathymetry field
+    bathymetry = np.zeros((lat.size, lon.size), dtype=np.float32)
+    bathymetry[:, :] = depth[-1]
+    data = {"bathymetry": bathymetry}
+    dimensions = {"lon": lon, "lat": lat}
+
+    fieldsetBathymetry = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat"
+    )
+
+    fieldset.add_field(fieldsetBathymetry.bathymetry)
+
+    # Add in an unbeaching field
+    unbeach_U = np.zeros((lat.size, lon.size), dtype=np.float32)
+    unbeach_V = np.zeros((lat.size, lon.size), dtype=np.float32)
+
+    unbeach_V[0,:] = 1.0 # Meridional unbeaching at the top and bottom of the domain
+    unbeach_V[-1,:] = -1.0
+
+    data = {"unbeach_U": unbeach_U, "unbeach_V": unbeach_V}
+    dimensions = {"lon": lon, "lat": lat}
+
+    fieldsetunbeaching = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat"
+    )
+
+    fieldset.add_field(fieldsetunbeaching.unbeach_U)
+    fieldset.add_field(fieldsetunbeaching.unbeach_V)
+
+
+    return fieldset
+
+def add_uniform_temp_salt_field(fieldset, times):
+    """ Add a uniform temperature and salinity field to the fieldset.
+    The temperature/salinity field is time-invariant and has a linear decay with depth.
+    Of course, this makes no sense in reality, but it is a simple example."""
+    lon = fieldset.U.grid.lon #TODO Given the Bickley Jet above is C-Grid, is this the right approach to create the other velocity fields??
+    lat = fieldset.U.grid.lat
+    depth = fieldset.U.grid.depth
+
+    T = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+    S = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+
+    T[:, 0, :, :] = 25.0 # TODO: make sure the units are correct!
+    S[:, 0, :, :] = 35.0
+
+    for d in range(1, depth.size):
+        T[:, d, :, :] = T[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
+        S[:, d, :, :] = S[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
```

Comment:
WHy these commented parts of the statements? Are they stray debug messages?

Reply:
I had a depth-varying temp/salt field, but I found it too cumbersome to test with, so I decided on a uniform in spacetime field. Will remove